src/components/Blog.js

```diff
@@ -0,0 +1,222 @@
+import React, { useEffect, useState } from 'react';
+import { hot } from 'react-hot-loader';
+import PropTypes from 'prop-types';
+import {
+  Grid,
+  Typography,
+  Card,
+  Box,
+  Link,
+  Paper,
+  Button,
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import QuestionRender from './QuestionRender';
+
+const useStyles = makeStyles({
+  cardcontent: {
+    '&:last-child': { paddingBottom: 16 },
+  },
+  markdown: {
+    '& p': { marginBottom: 0, marginTop: 0, overflowWrap: 'anywhere' },
+  },
+  white: {
+    '&.cardroot': {
+      backgroundColor: 'white',
+    },
+    '&.noAnswer': {
+      opacity: '70%',
+      fontStyle: 'italic',
+    },
+  },
+  gray: {
+    '&.cardroot': {
+      backgroundColor: '#F9F9F9',
+    },
+    '&.noAnswer': {
+      opacity: '70%',
+      fontStyle: 'italic',
+    },
+  },
+  unfinishedRequired: {
+    color: 'red',
+    '&.cardroot': {
+      borderColor: 'red',
+    },
+  },
+});
+
+const Blog = ({
+  questions, saveBlogAnswers, isReadOnly, programID,
+}) => {
+  const [editMode, setEditMode] = useState(false);
+  const [blogQuestions, setBlogQuestions] = useState([]);
+  const classes = useStyles();
+
+  useEffect(() => {
+    setBlogQuestions(questions);
+  }, [questions]);
+
+  const toggleEditMode = () => {
+    if (editMode) {
+      saveBlogAnswers(programID, blogQuestions);
+    }
+    setEditMode(!editMode);
+  };
+
+  const getClass = () => {
+    if (!isReadOnly && editMode) {
+      return classes.gray;
+    }
+    return classes.white;
+  };
+
+  const getRequiredClass = (required, answer) => {
+    if (required && !answer && !isReadOnly) {
+      return classes.unfinishedRequired;
+    }
+    return null;
+  };
+
+  return (
+    <Grid container direction="column">
+      {
+        questions.length === 0 && isReadOnly ? (
+          <Typography
+            variant="caption"
+            style={{ color: 'grey', fontStyle: 'italic' }}
+          >
+            No Blog Post Available
+          </Typography>
+        ) : (
+          <>
+            {
+                !isReadOnly ? (
```

Comment:
These seem to have indented 4 spaces

Reply:
I think I fixed them all now. That is how the ESLint formatter was formatting them for some reason.